### PR TITLE
circleci: pack orbs to allow structuring them

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/6025.

See https://circleci.com/docs/2.0/creating-orbs/#packing-a-config.

### CODE REVIEW

**Comment by kopiczko** [comment]
I requested reviews too early. Need to check why it doesn't work.

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,13 +7,19 @@ workflows:
...
+          requires:
+            - orb-tools/lint
+
       - orb-tools/publish-dev:
           checkout: true
           orb-path: catalog/orb.yaml
           orb-name: giantswarm/catalog
           publish-token-variable: CIRCLECI_DEV_API_TOKEN
           requires:
-            - orb-tools/lint
```
Comment (MarcelMue) [inline .circleci/config.yml:16]: Why no more lint? or does pack also lint?

Comment (MarcelMue) [reply .circleci/config.yml:16]: Nevermind I should concentrate.

**Review by MarcelMue — APPROVED** [review]

**Comment by kopiczko** [comment]
@MarcelMue do you know why it is `ci/circleci: publish Expected — Waiting for status to be reported`?

**Comment by MarcelMue** [comment]
They changed the naming - `publish-dev` was previously called publish. I think I had it pinned in the branch protection github setting but removed it now.

You should be good to go.

**Comment by kopiczko** [comment]
Thanks. FMI what did you pin?

**Comment by MarcelMue** [comment]
When you go to the project settings in github -> Branches -> Branch protection rules -> edit -> status checks pretty far down on the page.

**Comment by kopiczko** [comment]
Thanks

**Pull request merged**